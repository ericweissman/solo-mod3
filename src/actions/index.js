export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (error) => ({
  type: 'HAS_ERRORED',
  error
})

export const fetchArtistsSuccess = (artists, genre) => ({
  type: 'FETCH_ARTISTS_SUCCESS',
  artists,
  genre
})

export const getDeepHouseSuccess = (artists) => ({
  type: 'GET_DEEP_HOUSE_SUCCESS',
  artists
})

export const getFutureBassSuccess = (artists) => ({
  type: 'GET_FUTURE_BASS_SUCCESS',
  artists
})

export const getGlitchHopSuccess = (artists) => ({
  type: 'GET_GLITCH_HOP_SUCCESS',
  artists
})

export const getTrapSuccess = (artists) => ({
  type: 'GET_TRAP_SUCCESS',
  artists
})

export const addToFavorites = (artist) => ({
  type: "ADD_TO_FAVORITES",
  artist
})

export const removeFromFavorites = (id) => ({
  type: "REMOVE_FROM_FAVORITES",
  id
})

export const populateFavorites = (artists) => ({
  type: 'POPULATE_FAVORITES',
  artists
})

