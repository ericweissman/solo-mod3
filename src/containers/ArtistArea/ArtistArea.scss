.artist-area {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-left: 185px;
}