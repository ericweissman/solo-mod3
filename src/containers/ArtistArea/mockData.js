export const mockPropsEmpty = {
  glitchHop: [],
  trap: [],
  deepHouse: [],
  futureBass: [],
  isLoading: false,
  favorites: [],
  fetchArtists: jest.fn()
}
export const mockPropsFetchedNoFaves = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: [],
  match: { path: '/glitchHop' },
  fetchArtists: jest.fn()
}
export const mockPropsFetchedWithFaves = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/glitchHop' },
  fetchArtists: jest.fn()
}
export const glitchHop = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/glitchHop' },
  fetchArtists: jest.fn()
}
export const deepHouse = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/deepHouse' },
  fetchArtists: jest.fn()
}
export const futureBass = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/futureBass' },
  fetchArtists: jest.fn()
}
export const trap = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/trap' },
  fetchArtists: jest.fn()
}
export const favorites = {
  glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/favorites' },
  fetchArtists: jest.fn()
}

export const emptyGH = {
  glitchHop: [],
  trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/glitchHop' },
  fetchArtists: jest.fn()
}
export const emptyTrap = {
  glitchHop: [],
  trap: [],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/trap' },
  fetchArtists: jest.fn()
}
export const emptyDH = {
  glitchHop: [],
  trap: [],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/trap' },
  fetchArtists: jest.fn()
}
export const emptyFB = {
  glitchHop: [],
  trap: [],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '/trap' },
  fetchArtists: jest.fn()
}
export const emptyPath = {
  glitchHop: [],
  trap: [],
  deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
  futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
  isLoading: false,
  favorites: ['1234'],
  match: { path: '' },
  fetchArtists: jest.fn()
}