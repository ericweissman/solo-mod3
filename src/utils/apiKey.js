 const apiKey = `330926-musicly-0PJ7J0DT`

 export default apiKey