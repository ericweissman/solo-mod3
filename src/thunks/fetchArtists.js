import {isLoading, hasErrored, fetchArtistsSuccess} from '../actions'

export const fetchArtists = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(fetchArtistsSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}