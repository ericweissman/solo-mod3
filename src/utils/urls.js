import apiKey from './apiKey'
export const deepHouse = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=robin schulz&k=${apiKey}&type=band&verbose=1&info=1&limit=15`
export const glitchHop = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=michal menert&k=${apiKey}&type=band&verbose=1&info=1&limit=15`
export const futureBass = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=hermitude&k=${apiKey}&type=band&verbose=1&info=1&limit=15`
export const trap = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=baauer&k=${apiKey}&type=band&verbose=1&info=1&limit=15`