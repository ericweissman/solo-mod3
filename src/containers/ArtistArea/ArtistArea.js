import React, { Component } from 'react'
import * as actions from '../../actions'
import { fetchArtists } from '../../thunks/fetchArtists'
import { connect } from 'react-redux'
import Card from '../../components/Card/Card'
import * as url from '../../utils/urls'
import Loading from '../../components/Loading/Loading'
const key = require('short-id')

class ArtistArea extends Component {

  getArtists = () => {
    switch (this.props.match.path) {
      case '/glitchHop':
        if (this.props.glitchHop.length === 0) {
          this.props.fetchArtists(url.glitchHop, actions.getGlitchHopSuccess)
        }
        return 'glitchHop'
      case '/deepHouse':
        if (this.props.deepHouse.length === 0) {
          this.props.fetchArtists(url.deepHouse, actions.getDeepHouseSuccess)
        }
        return 'deepHouse'
      case '/trap':
        if (this.props.trap.length === 0) {
          this.props.fetchArtists(url.trap, actions.getTrapSuccess)
        }
        return 'trap'
      case '/futureBass':
        if (this.props.futureBass.length === 0) {
          this.props.fetchArtists(url.futureBass, actions.getFutureBassSuccess)
        }
        return 'futureBass'
      case '/favorites':
        return 'favorites'
      default:
        return null
    }
  }

  cardsToDisplay = () => {
    const genre = this.getArtists()
    const cards = this.props[genre].map(artist => {
      return <Card artist={artist} key={key.generate()} />
    })
    const {isLoading} = this.props
    if (isLoading) {
      return <Loading />
    }
    return cards
  }

  render() {
    return (
      <div>
        {this.cardsToDisplay()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  glitchHop: state.glitchHop,
  trap: state.trap,
  deepHouse: state.deepHouse,
  futureBass: state.futureBass,
  isLoading: state.isLoading,
  favorites: state.favorites,
})

export const mapDispatchToProps = (dispatch) => ({
  fetchArtists: (url, actionToDispatch) => dispatch(fetchArtists(url, actionToDispatch))
})

export default connect(mapStateToProps, mapDispatchToProps)(ArtistArea)