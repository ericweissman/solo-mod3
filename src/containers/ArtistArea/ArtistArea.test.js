import React from 'react'
import { shallow } from 'enzyme'
import {fetchArtists} from '../../thunks/fetchArtists'
import * as actions from '../../actions'
import * as url from '../../utils/urls'
import {mockPropsFetchedWithFaves, glitchHop, deepHouse, futureBass, trap, favorites, emptyGH } from '../ArtistArea/mockData'
import { ArtistArea, mapDispatchToProps, mapStateToProps } from '../ArtistArea/ArtistArea'

describe('ArtistArea', () => {
  let wrapper

  describe('snapshots', () => {
    it('should match the correct snapshot', () => {
      wrapper = shallow(<ArtistArea {...glitchHop} />)
      expect(wrapper).toMatchSnapshot()
    })
    it('should match the correct snapshot', () => {
      wrapper = shallow(<ArtistArea {...deepHouse} />)
      expect(wrapper).toMatchSnapshot()
    })
    it('should match the correct snapshot', () => {
      wrapper = shallow(<ArtistArea {...futureBass} />)
      expect(wrapper).toMatchSnapshot()
    })
    it('should match the correct snapshot', () => {
      wrapper = shallow(<ArtistArea {...trap} />)
      expect(wrapper).toMatchSnapshot()
    })
    it('should match the correct snapshot', () => {
      wrapper = shallow(<ArtistArea {...favorites} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('getArtists', () => {
    let wrapper
    it('should fetch the correct data if there is no data', () => {
      wrapper = shallow(<ArtistArea test={test} {...emptyGH} />)
      wrapper.instance().getArtists()
      expect(wrapper.prop('test')).toEqual(false)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with the correct properties', () => {
      const mockState = {
        glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
        trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
        deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
        futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
        isLoading: false,
        favorites: ['1234'],
        testProp: 'fake',
        testProp2: 'also fake'
      }
      const expected = {
        glitchHop: [{ name: 'test', id: '1234', wiki: 'asfas', favorited: false }],
        trap: [{ name: 'test', id: '22', wiki: 'asfas', favorited: false }],
        deepHouse: [{ name: 'test', id: '33', wiki: 'asfas', favorited: false }],
        futureBass: [{ name: 'test', id: '44', wiki: 'asfas', favorited: false }],
        isLoading: false,
        favorites: ['1234'],
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with the correct action when there has not been a fetch', () => {
      wrapper = shallow(<ArtistArea {...emptyGH} />)
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchArtists(url.glitchHop, actions.getGlitchHopSuccess)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchArtists(url.glitchHop, actions.getGlitchHopSuccess)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

})