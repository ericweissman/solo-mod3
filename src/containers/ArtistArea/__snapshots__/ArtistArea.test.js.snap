// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistArea snapshots should match the correct snapshot 1`] = `
<div>
  <Connect(Card)
    artist={
      Object {
        "favorited": true,
        "id": "1234",
        "name": "test",
        "wiki": "asfas",
      }
    }
    key="02f6d4"
  />
</div>
`;

exports[`ArtistArea snapshots should match the correct snapshot 2`] = `
<div>
  <Connect(Card)
    artist={
      Object {
        "favorited": false,
        "id": "33",
        "name": "test",
        "wiki": "asfas",
      }
    }
    key="786dfc"
  />
</div>
`;

exports[`ArtistArea snapshots should match the correct snapshot 3`] = `
<div>
  <Connect(Card)
    artist={
      Object {
        "favorited": false,
        "id": "44",
        "name": "test",
        "wiki": "asfas",
      }
    }
    key="bcc274"
  />
</div>
`;

exports[`ArtistArea snapshots should match the correct snapshot 4`] = `
<div>
  <Connect(Card)
    artist={
      Object {
        "favorited": false,
        "id": "22",
        "name": "test",
        "wiki": "asfas",
      }
    }
    key="f31661"
  />
</div>
`;

exports[`ArtistArea snapshots should match the correct snapshot 5`] = `
<div>
  <Connect(Card)
    artist={
      Object {
        "favorited": false,
        "id": "1234",
        "name": "test",
        "wiki": "asfas",
      }
    }
    key="024b5f"
  />
</div>
`;
